pandas/_libs/period.pyx

```diff
@@ -33,9 +33,12 @@ from tslibs.timezones cimport (
     is_utc, is_tzlocal, get_utcoffset, get_dst_info, maybe_get_tz)
 from tslibs.timedeltas cimport delta_to_nanoseconds
 
-from tslibs.parsing import (parse_time_string, NAT_SENTINEL,
-                            _get_rule_month, _MONTH_NUMBERS)
-from tslibs.frequencies cimport get_freq_code
+from tslibs.frequencies cimport (get_freq_code, get_base_alias,
+                                 get_to_timestamp_base, _get_freq_str,
+                                 _get_rule_month)
+from tslibs.frequencies import _MONTH_NUMBERS
```

Comment:
import from ccalendar?

Reply:
If/when `ccalendar` is implemented absolutely.  ATM it looks like `ccalendar` may entail a non-trivial perf hit, so that part of #18489 may need to be reverted.  It's a shame -- deps would be a lot simpler if `is_leapyear`, `weekday`, `days_in_month` didn't require all of `src/datetime`.